### PR TITLE
feat: show password improvement 

### PR DESCRIPTION
**Pull request by martaradziszewska**
# Related issue
Closes: #517

# Scope of work
- Hide/Show Password functionality to SfInput 
- Added slot with SfIcon as a button with simple animation of switching password visibility 
- Created story for this case

# Screenshots of visual changes
Peek2 2020-01-06 18-19

# Checklist

- [x] I followed [composition rules](https://docs.storefrontui.io/contributing/coding-guidelines.html#component-rules) for my component
- [x] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/vue/src/components/atoms/SfInput/SfInput.html [hunk 1]
```diff
@@ -20,6 +20,16 @@
       <!-- @slot Custom input label -->
       <slot name="label" v-bind="{label}">{{ label }}</slot>
     </label>
+    <template v-if="isPasswordType">
+        <!-- @slot -->
+        <slot name="show-password-button" v-bind="{ isPasswordVisible, switchVisibilityPassword, isPasswordType }">
```
Comment (aniamusial) [inline packages/vue/src/components/atoms/SfInput/SfInput.html]: Its a bit long, maybe simply show-password instead? :)

File: packages/vue/src/components/atoms/SfInput/SfInput.stories.js [hunk 2]
```diff
@@ -128,4 +128,41 @@ storiesOf("Atoms|Input", module)
         <SfIcon icon="info_shield" size="10px" color="#E22326" style="margin-right: 4px; display: inline-block"/> CUSTOM ERROR MESSAGE
       </template>
     </SfInput>`
+  }))
+  .add("[slot] password visibility icon", () => ({
```
Comment (aniamusial) [inline packages/vue/src/components/atoms/SfInput/SfInput.stories.js]: Change to match slot name

File: packages/vue/public/assets/show_password.svg [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<svg width="22" height="13" viewBox="0 0 22 13" fill="none" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (martaradziszewska) [reply packages/vue/public/assets/show_password.svg]: there is change in icons.js on the top of diff files :top:

File: packages/shared/icons/icons.js [hunk 4]
```diff
@@ -149,6 +149,14 @@ export const icons = {
     paths: [
       "M10.9506 28C14.5861 23.8734 21.4626 16.112 21.4626 10.4135C21.4626 4.71492 16.7463 0 10.9506 0C5.25203 0 0.537109 4.71632 0.537109 10.4135C0.537109 16.112 7.41447 23.8731 10.9506 28ZM10.9506 6.09112C13.4071 6.09112 15.3715 8.05549 15.3715 10.4138C15.3715 12.8703 13.4071 14.8347 10.9506 14.8347C8.59246 14.8347 6.62795 12.8703 6.62795 10.4138C6.62795 8.05563 8.59232 6.09112 10.9506 6.09112Z"
     ]
+  },
+  show_password: {
```
Comment (martaradziszewska) [inline packages/shared/icons/icons.js]: @aniamusial here :)

File: packages/shared/styles/components/SfInput.scss [hunk 5]
```diff
@@ -109,6 +118,41 @@ $input__error-message-font-size: $font-size-small-mobile !default;
     font-family: $input__error-message-font-family;
     font-size: $input__error-message-font-size;
   }
+  &__password-button {
+    display: block;
+    position: absolute;
```
[the remark this answers is not in the document]
Comment (martaradziszewska) [reply packages/shared/styles/components/SfInput.scss]: thanks! thats better way indeed

File: packages/vue/src/components/atoms/SfInput/SfInput.html [hunk 6]
```diff
@@ -5,9 +5,9 @@
     'sf-input--invalid': valid === false
   }"
 >
-  <div style="position: relative;">
+  <div class="sf-input__wrapper">
     <input
-      v-bind="$attrs"
+      v-bind="attrs"
```
[the remark this answers is not in the document]
Comment (martaradziszewska) [reply packages/vue/src/components/atoms/SfInput/SfInput.html]: to create an instance of defined attributes in result we can have access and modify them in computed property. In this solution we can change type of input from `password` to `text` to see password without changing its original type in component

File: packages/vue/public/assets/show_password.svg (hunk 3 above)
[the remark this answers is not in the document]
Comment (aniamusial) [reply packages/vue/public/assets/show_password.svg]: Ok, so this one is not needed in the project, you can safely delete it :)

File: packages/vue/src/components/atoms/SfInput/SfInput.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (aniamusial) [reply packages/vue/src/components/atoms/SfInput/SfInput.html]: I was just wondering if there is a more direct way to access attrs than returning in computed property? But in general it looks good to me :) WDYT @pspaczek ?

File: packages/vue/src/components/atoms/SfInput/SfInput.vue [hunk 7]
```diff
@@ -21,6 +21,36 @@
         <!-- @slot Custom input label -->
         <slot name="label" v-bind="{ label }">{{ label }}</slot>
       </label>
+      <template v-if="isPasswordType">
+        <!-- @slot -->
+        <slot
```
Comment (pspaczek) [inline packages/vue/src/components/atoms/SfInput/SfInput.vue]: imo you use here to many slots
```
<slot name="show-password" v-if="isPassword" v-bind={}>
  <SfButton />
</slot>
```

**Comment by martaradziszewska** [comment]
@pspaczek In creating this solution I don't want to use props and copy value of this props to local state. Instead I want to get type from what we already have available to use -> `this.$attrs` and create a copy  -> `attrs()` - this is not a prop, this is a copy of dom attrs. As a result, this solution gives the same effect as you proposed -> changing type not changing its defined type. I hope I explain my solution clearly. I changed it as you proposed -> its more readable to understand :) thanks

**Pull request merged**